File: day71-80/day76/2044. Count Number of Maximum Bitwise-OR Subsets.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var countMaxOrSubsets = function (nums) {
  function getCombinations(arr, r) {
    const results = [];

    function combine(start, combination) {
      if (combination.length === r) {
        results.push([...combination]);
        return;
      }

      for (let i = start; i < arr.length; i++) {
        combination.push(arr[i]);
        combine(i + 1, combination);
        combination.pop();
      }
    }

    combine(0, []);
    return results;
  }
  let max_or = 0;
  let count = 0;
  for (let num of nums) {
    max_or |= num;
  }

  for (let r = 1; r <= nums.length; r++) {
    for (let comb of getCombinations(nums, r)) {
      let or = 0;
      for (let num of comb) {
        or |= num;
      }
      if (or === max_or) {
        count++;
      }
    }
  }
  return count;
};

console.log(countMaxOrSubsets([3, 2, 1, 5]));

File: day71-80/day76/2044. Count Number of Maximum Bitwise-OR Subsets.py
from typing import List
from itertools import combinations
class Solution:
    def countMaxOrSubsets(self, nums: List[int]) -> int:
        max_or = 0
        count = 0
        for num in nums:
            max_or |= num
        
        for r in range(1, len(nums) + 1):
            for comb in combinations(nums, r):
                subset_or = 0
                for num in comb:
                    subset_or |= num
                if subset_or == max_or:
                    count += 1
        return count
    
sol=Solution()
nums=[2,2,2]

print(sol.countMaxOrSubsets(nums))